### ISSUE TITLE
[yugabyted] [QA] [Suggestion] Unencrypted password for secure connections in logs, console

### ISSUE THREAD

**Issue by zlareb1-yb** [yugabyte/yugabyte-db#14112]
### Description

Same password is being used for Postgress, YCQL, YSQL and we show unencrypted password in console, logs & yugabyted_credentials.txt

**Console**:
```
MacBook-Pro-16-inch-2021:yugabyte-2.15.3.0 mzafar$ ./bin/yugabyted start --secure --certs_dir ~/certs/127.0.0.1 --advertise_address=127.0.0.1
Starting yugabyted...
✅ System checks           
✅ YugabyteDB Started                  
✅ SSL Authentication enabled.   
✅ User Authentication enabled.
YugabyteDB Started

+----------------------------------------------------------------------------------------------------------+
|                                                yugabyted                                                 |
+----------------------------------------------------------------------------------------------------------+
| Status              : Running.                                                                           |
| Replication Factor  : 1                                                                                  |
| Web console         : http://127.0.0.1:7000                                                              |
| JDBC                : jdbc:postgresql://127.0.0.1:5433/yugabyte?user=yugabyte&password=2FCAvI1jDZPn      |
| YSQL                : bin/ysqlsh   -U yugabyte -d yugabyte                                               |
| YCQL                : bin/ycqlsh   -u cassandra --ssl                                                    |
| Data Dir            : /Users/mzafar/var/data                                                             |
| Log Dir             : /Users/mzafar/var/logs                                                             |
| Universe UUID       : e89f6cfb-79a1-4cdd-968c-2095010e62cd                                               |
+----------------------------------------------------------------------------------------------------------+
```

**yugabyted_credentials.txt**:
```
MacBook-Pro-16-inch-2021:yugabyte-2.15.3.0 mzafar$ cat /Users/mzafar/var/data/yugabyted_credentials.txt
YSQL Credentials:
Username: yugabyte
Password: 2FCAvI1jDZPn
Database: yugabyte

YCQL Credentials:
Username: cassandra
Password: 2FCAvI1jDZPn
Keyspace: None
```

**Logs**:
**Also the password is available in logs**:
`[yugabyted start] 2022-09-21 19:40:04,432 INFO:  | 65.2s | run_process: cmd: [u'/Users/mzafar/ybd/yugabyte-2.15.3.0/bin/ysqlsh', u'postgresql://yugabyte:yugabyte@127.0.0.1:5433/yugabyte', u'-c', u"alter role yugabyte password 'lH76UmClWDZu';"]`

Should we be encrypting the password here?

**Comment by nchandrappa** [comment]
This will be a security enhancement request. We will need to obfuscate the passwords when writing to the logs.

**Comment by gargsans-yb** [comment]
As discussed with @nchandrappa , passwords showing in console output is a valid behaviour. Changes committed to hide the password from logs.
Landed on master and Backported to 2.17.0
